### PR TITLE
Add documentation for Transfer APIs

### PR DESCRIPTION
**Pull request by duduribeiro**
This PR adds api doc and openapi spec for getDomainTransfer and cancelDomainTransfer

### CODE REVIEW

File: fixtures/v2/api/getDomainTransfer/success.http [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+X-Request-Id: 4818d867-1f72-4619-ab46-d345e6dd25eb
+X-Runtime: 0.092854
+Strict-Transport-Security: max-age=31536000
+
+{
+  "data": {
+    "id": 1,
+    "domain_id": 1,
+    "registrant_id": 1,
+    "state": "cancelled",
```
Comment (weppos) [inline fixtures/v2/api/getDomainTransfer/success.http:22]: It turns out the transfer object doesn't have this state documented.

```
    DomainTransfer:
      type: object
      properties:
        id:
          type: integer
        domain_id:
          type: integer
        registrant_id:
          type: integer
        state:
          type: string
          enum:
            - new
            - transferring
            - transferred
            - failed
```

Can you please update it? It's also possible the DomainRegistration object is affected.

**Comment by weppos** [comment]
@duduribeiro I am removing myself as reviewer for now. We are going to keep this branch open to document all the transfer API.

**Comment by duduribeiro** [comment]
@weppos re-adding you as reviewer, since I added the documentations for Show and Cancel Domain Transfer.

**Comment by duduribeiro** [comment]
@weppos updated

**Comment by weppos** [comment]
@duduribeiro I am removing myself as reviewer. We have more work to do on this project, and we can continue to update this PR without triggering a review on each change. Please don't add me again until we have finalized all the work and we are ready to ship the docs. Once ready, we'll make a final, single review.

**Comment by duduribeiro** [comment]
@weppos I'm adding you as reviewer, as I think the API side is finished. We already exposed the field on it.

**Pull request merged**